transformations/transformations/single_column_coalesced_vector.py

```diff
@@ -193,6 +199,41 @@ def process_kernel(self, routine):
         section_mapper = {s: ir.Section(body=s, label='vector_section') for s in sections}
         routine.body = NestedTransformer(section_mapper).visit(routine.body)
 
+    def process_driver(self, routine, targets=()):
+        """
+        Applies the SCCDevector utilities to a "driver". This consists simply
+        of stripping vector loops and determining which sections of the IR can be
+        placed within thread-parallel loops.
+
+        Parameters
+        ----------
+        routine : :any:`Subroutine`
+            Subroutine to apply this transformation to.
+        targets : list or string
+            List of subroutines that are to be considered as part of
+            the transformation call tree.
+        """
+
+        with pragmas_attached(routine, ir.Loop, attach_pragma_post=True):
+            driver_loops = SCCBaseTransformation.find_driver_loops(routine=routine, targets=targets)
+
+        # remove vector loops
+        driver_loop_map = {}
+        for loop in driver_loops:
+            loop_map = {}
+            for l in FindNodes(ir.Loop).visit(loop.body):
+                if l.variable == self.horizontal.index:
+                    loop_map[l] = l.body
+            new_driver_loop = Transformer(loop_map).visit(loop.body)
+            new_driver_loop = loop.clone(body=new_driver_loop)
+            sections = self.extract_vector_sections(new_driver_loop.body, self.horizontal)
+            if self.trim_vector_sections:
+                sections = self.get_trimmed_sections(routine, self.horizontal, sections)
```

Comment:
Have you tested this with `trimmed_vector_sections` enabled? I believe in the current guise of the dataflow analysis this will give a runtime error. #158 tries to update the dataflow analysis to fix this.

Reply:
Once that is in, I believe you should be able to remove `!$loki separator` pragmas.

Reply:
Tested it with and without `self.get_trimmed_sections(...)`. However, (to be honest) haven't checked whether it actually makes a difference (for the code parts I used it for). Regarding the removal `!$loki separator` we should probably make this another PR once confirmed that it works (considering possible edge cases)?!